pymc3/stats.py

```diff
@@ -59,44 +60,60 @@ def wrapped_f(pymc3_obj, *args, **kwargs):
 
 
 @statfunc
-def autocorr(x, lag=1):
-    """Sample autocorrelation at specified lag.
+def autocorr(x, lag=None):
+    """
+    Compute autocorrelation using FFT for every lag for the input array
+    https://en.wikipedia.org/wiki/Autocorrelation#Efficient_computation
 
     Parameters
     ----------
     x : Numpy array
         An array containing MCMC samples
-    lag : int
-        The desidered lag to take in consideration
+
+    Returns
+    -------
+    acorr: Numpy array same size as the input array
     """
-    S = autocov(x, lag)
-    return S[0, 1] / np.sqrt(np.prod(np.diag(S)))
+    y = x - x.mean()
```

Comment:
is this why we don't `x -= x.mean()`? maxim caught me doing that in `imcmc` last week...

Reply:
Yep, took me awhile to find out because it silently modifies the trace... the computed value is always off by a little bit but some is super close - drives me crazy for a day :-/